xtext/org.icyphy.linguafranca/src/org/icyphy/generator/GeneratorBase.xtend

```diff
@@ -918,74 +988,56 @@ abstract class GeneratorBase extends AbstractLinguaFrancaValidator {
-            builder.command(bashCommand)
-            val stdout = new ByteArrayOutputStream()
-            val stderr = new ByteArrayOutputStream()
-            val returnCode = builder.runSubprocess(stdout, stderr)
-            if (stdout.size() > 0) {
-                println("--- Standard output from command:")
-                println(stdout.toString())
-                println("--- End of standard output.")
-            }
-            if (stderr.size() > 0) {
-                println("--- Standard error from command:")
-                println(stderr.toString())
-                println("--- End of standard error.")
-            }
-            
-            if (returnCode !== 0) {
-                if (mode === Mode.INTEGRATED) {
-                    reportCommandErrors(stderr.toString())
-                } else {
-                    reportError("Bash command returns error code " + returnCode)
-                }
-            }
-            return returnCode
+     * @param cmd The command to be executed
+     * @return A ProcessBuilder object if the command was found or null otherwise.
+     */
+    protected def createCommand(String cmd) {
+        return createCommand(cmd, #[])
+    }
+    
+    /**
+     * Creates a ProcessBuilder for a given command and its arguments.
+     * 
+     * This method ensures that the given command is executable,
+     * It first tries to find the command with 'which cmake'. If that
+     * fails, it tries again with bash. n case this fails again,
+     * it returns null. Otherwise, a correctly constructed ProcessBuilder
+     * object is returned. 
+     * 
+     * @param cmd The command to be executed
+     * @param args A list of arguments for the given command
+     * @return A ProcessBuilder object if the command was found or null otherwise.
+     */
+    protected def createCommand(String cmd, List<String> args) {
+        // Make sure the command is found in the PATH.
+        print('''--- Looking for command «cmd» ... ''')
+        val whichBuilder = new ProcessBuilder(#["which", cmd])
+        val whichReturn = whichBuilder.start().waitFor()
+        if (whichReturn == 0) {
+            println("SUCCESS")
+            val builder = new ProcessBuilder(#[cmd] + args)
+            builder.directory(new File(directory))
+            return builder
         }
+        println("FAILED")
+        // Try running with bash.
+        // The --login option forces bash to look for and load the first of
+        // ~/.bash_profile, ~/.bash_login, and ~/.bashrc that it finds.
+        print('''--- Trying again with bash ... ''')
+        val bashCommand = #["bash", "--login", "-c", '''which «cmd»''']
+        val bashBuilder = new ProcessBuilder(bashCommand)
+        val bashReturn = bashBuilder.start().waitFor()
+        if (bashReturn == 0) {
+            println("SUCCESS")
+            val builder = new ProcessBuilder(#["bash", "--login", "-c", '''«cmd» «args.join(" ")»'''])
```

Comment:
Why invoke the compiler using Bash? If we invoke `which` in Bash it will give us the full path to the command that we can execute directly. Why not use that? It will be less confusing when the actual compile command is shown upon encountering an error.

Reply:
Right. Good idea! That thought simply did not occur to me. I will give that a try